dascore/utils/misc.py

```
@@ -193,7 +194,53 @@ def _iter_filesystem(
     """
+
+    def _iter_generic(path_like):
+        warned_no_remote_mtime = False
+
+        def _meets_timestamp(entry):
+            nonlocal warned_no_remote_mtime
+            if timestamp is None:
+                return True
+            try:
+                return entry.stat().st_mtime >= timestamp
+            except Exception:
```

Ruff (0.15.7)

[warning] Do not catch blind exception: `Exception`

(BLE001)